File: app/assets/javascripts/DataRequirement.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const DataRequirementSchema = DataElementSchema({
  type: Any,
  profile: Any,
  subject: Any,
  mustSupport: Any,
  codeFilter: Any,
  dateFilter: Any,
  limit: Any,
  sort: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'DataRequirement' },

});

module.exports.DataRequirementSchema = DataRequirementSchema;
module.exports.DataRequirement = mongoose.model('DataRequirement', DataRequirementSchema);

File: app/models/qdm/data_requirement.rb
module QDM
  # app/models/qdm/data_requirement.rb
  class DataRequirement < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :type
    field :profile
    field :subject
    field :mustSupport
    field :codeFilter
    field :dateFilter
    field :limit
    field :sort
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
